File: src/paths.ts
import { homedir } from 'node:os';
import { join } from 'node:path';

/**
 * Every directory aitrack owns under the user's home.
 *
 * These were previously rebuilt from `homedir()` in four separate modules, so
 * moving the application directory meant finding all four. They are evaluated
 * at module load time: the test suite mocks `node:os` before importing anything
 * that reaches these, and making them lazy would break that arrangement.
 */
export const APP_DIR = join(homedir(), '.config', 'aitrack');

export const CONFIG_PATH = join(APP_DIR, 'config.json');

/** Clone of the user's data repo. */
export const LOCAL_REPO = join(APP_DIR, 'repo');

/** Machine JSON files inside the clone. */
export const DATA_DIR = join(LOCAL_REPO, 'data');

/** Spool for machine files written while the repo was unavailable. */
export const PENDING_DATA_DIR = join(APP_DIR, 'pending', 'data');

/** Parsed-transcript cache, keyed by source mtime and size. */
export const CACHE_DIR = join(APP_DIR, 'cache');

File: crates/aitrack-core/src/paths.rs
//! Application directories under the user's home.
//!
//! Mirrors `packages/aitrack-lib/src/paths.ts`: always `~/.config/aitrack`.

use std::path::PathBuf;
use std::sync::OnceLock;

use crate::env::environment_value;

fn home_dir() -> PathBuf {
    if let Some(override_home) = environment_value("HOME") {
        return PathBuf::from(override_home);
    }
    if let Some(profile) = environment_value("USERPROFILE") {
        return PathBuf::from(profile);
    }
    PathBuf::from(".")
}

static APP_DIR: OnceLock<PathBuf> = OnceLock::new();

fn app_dir_cached() -> &'static PathBuf {
    APP_DIR.get_or_init(|| home_dir().join(".config").join("aitrack"))
}

pub fn app_dir() -> PathBuf {
    app_dir_cached().clone()
}

pub fn config_path() -> PathBuf {
    app_dir().join("config.json")
}

pub fn local_repo() -> PathBuf {
    app_dir().join("repo")
}

pub fn data_dir() -> PathBuf {
    local_repo().join("data")
}

pub fn pending_data_dir() -> PathBuf {
    app_dir().join("pending").join("data")
}

pub fn cache_dir() -> PathBuf {
    app_dir().join("cache")
}
